nlp4all/models/Data.py

```diff
@@ -0,0 +1,29 @@
+# pylint: disable=invalid-name
+"""Data.py: Data model
+
+Data imported into nlp4all, used in analyses
+"""
+
+from __future__ import annotations
+
+from sqlalchemy import String, ForeignKey
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from .database import Base, NestedMutableJSONB
+
+from . import DataSource, DataAnnotation, DataTag, DataTagCategory
+
+class Data(Base): # pylint: disable=too-few-public-methods
+    """Data class for uploaded data"""
+
+    __tablename__ = 'nlp_data'
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    data_source_id: Mapped[int] = mapped_column(ForeignKey("data_source.id"))
+    data_source: Mapped[DataSource] = relationship(back_populates="data")
+    text: Mapped[str] = mapped_column(String(500)) # does this need to be longer?
```

Comment:
if we use meta on datasource, text can be deleted and replaced with a __get__()

Data.text -> Data.document...[DataSource.metadata.text_path]] or something similar...

This allows for virtually unlimited length main text fields, and avoids having to copy the main text from each tweet into the column "text" which uses unnecessary space, and we also don't need to care if something is a tweet, or a page from a book (i.e. how long should the String column be, 50, 500, 5000000)...

This would still allow for a consistent interface from a programming and conceptual perspective, by leaving the "text" property but making it a virtual pointer rather than a separate column

Reply:
finally, to aid filtering, it might be worth keeping some metadata ranges of identified date fields (only ones the users decide to keep), or things like that? (maybe i need to put a datetime schema node in the json schema generation!)

Reply:
done